aten/src/ATen/common_with_cwrap.py

```diff
@@ -30,8 +30,14 @@ def set_declaration_defaults(declaration):
         declaration['cname'] = declaration['name']
     if 'backends' not in declaration:
         declaration['backends'] = ['CPU', 'CUDA']
-    if 'api_name' not in declaration:
-        declaration['api_name'] = declaration['name']
+    assert 'api_name' not in declaration
+    declaration['api_name'] = declaration['name']
+    # NB: keep this in sync with gen_autograd.py
+    if declaration.get('overload_name'):
+        declaration['type_wrapper_name'] = "{}_{}".format(
```

Comment:
you can do this if you want, but for state like this that is totally derivable, I think it's clearer if you:
1) don't put the type_wrapper_name in the declaration, use a common function to compute it
2) don't use the environment, just pass the key explicitly to the CodeTemplate
(this is in the line with what @bhosmer did for gen_python_functions).

I do think this is okay given we haven't "cleaned up" function_wrapper yet, and this is in the existing style.